### PR TITLE
Changes body collision to knockdown, adds a throw_range of 4 to carbon/human

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do
All body collisions from being thrown now do 2 seconds of knockdown instead of stun, but brute damage is based on how fast you're thrown to compensate. A collision from when the thrower is standing still does 15 damage, if they are walking (with an aggressive grab so slowly) it does 18 damage (it was previously 10 damage). It does no stamina damage.

This PR also adds a throw_range of 4 to any carbon/humans, meaning people cant be thrown across your entire screen.

This also slightly changes up how gravitational anomalies work, since you cant get stunned by them but they are much more deadly with the brute damage increase.

## Why It's Good For The Game
Removes hard stun in favor of knockdown/stamina combat.

## Changelog
:cl:
tweak: Colliding into something when thrown will now make you knockdowned instead of stunned
add: Brute damage from a collision is now based on the thrown person's speed (a running/walking start before throwing someone will increase damage slightly)
tweak: you can now only throw humanoids 4 tiles instead of 7
soundadd: throwing someone into a wall will make a sound effect
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -591,17 +591,19 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
...
 		var/mob/living/carbon/victim = hit_atom
 		if(victim.flying)
 			return
 		if(hurt)
-			victim.take_organ_damage(10)
-			take_organ_damage(10)
-			victim.Weaken(2 SECONDS)
-			Weaken(2 SECONDS)
+			victim.take_organ_damage(7.5*speed)// speed while thrower is standing still is 2, while walking with an aggressive grab is 2.4
+			take_organ_damage(7.5*speed)
```
Comment (Sirryan2002) [inline code/modules/mob/living/carbon/carbon.dm:604]:
```suggestion
			take_organ_damage(7.5 * speed)
```

File: code/modules/mob/living/carbon/carbon.dm [hunk 2]
```diff
@@ -591,17 +591,19 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
...
 	if(iscarbon(hit_atom) && hit_atom != src)
 		var/mob/living/carbon/victim = hit_atom
 		if(victim.flying)
 			return
 		if(hurt)
-			victim.take_organ_damage(10)
-			take_organ_damage(10)
-			victim.Weaken(2 SECONDS)
-			Weaken(2 SECONDS)
+			victim.take_organ_damage(7.5*speed)// speed while thrower is standing still is 2, while walking with an aggressive grab is 2.4
```
Comment (Sirryan2002) [inline code/modules/mob/living/carbon/carbon.dm:603]:
```suggestion
			victim.take_organ_damage(7.5 * speed)// speed while thrower is standing still is 2, while walking with an aggressive grab is 2.4
```

File: code/modules/mob/living/carbon/carbon.dm [hunk 3]
```diff
@@ -591,17 +591,19 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
 				hurt = FALSE*/
 	if(hit_atom.density && isturf(hit_atom))
 		if(hurt)
-			Weaken(2 SECONDS)
-			take_organ_damage(10)
+			visible_message("<span class='danger'>[src] slams into [hit_atom]!</span>", "<span class='userdanger'>You slam into [hit_atom]!</span>")
+			take_organ_damage(7.5*speed) // speed while thrower is standing still is 2, while walking with an aggressive grab is 2.4
```
[the remark this answers is not in the document]
Comment (Contrabang) [reply code/modules/mob/living/carbon/carbon.dm:595]: Tested, it only does 15/18 damage to a single limb. Minimum is 30 damage to break a limb (it does no internal damage btw, that proc is just named weird)

File: code/modules/mob/living/carbon/carbon.dm [hunk 4]
```diff
@@ -591,17 +591,19 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
...
 		var/mob/living/carbon/victim = hit_atom
 		if(victim.flying)
 			return
 		if(hurt)
-			victim.take_organ_damage(10)
-			take_organ_damage(10)
-			victim.Weaken(2 SECONDS)
-			Weaken(2 SECONDS)
+			victim.take_organ_damage(7.5 * speed)// speed while thrower is standing still is 2, while walking with an aggressive grab is 2.4
+			take_organ_damage(7.5 * speed)
```
Comment (hal9000PR) [inline code/modules/mob/living/carbon/carbon.dm:604]:
```suggestion
			var/damage = 7.5 * speed
			victim.take_organ_damage(damage) // speed while thrower is standing still is 2, while walking with an aggressive grab is 2.4
			take_organ_damage(damage)
```
no need to do the maths twice.

**Review by hal9000PR — CHANGES_REQUESTED** [review]
mirroring this from the discord. the damage being 7.5 * speed is not gonna fly in its current state. some fairly accessible things have really high throw speeds, crashing waves kick, the carp combo, has a throw speed of **14** meaning that this will do **105** damage. 
I suggest something more like `damage = 10 + 1.5 * speed`

File: code/modules/mob/living/carbon/carbon.dm [hunk 5]
```diff
@@ -591,17 +591,19 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
 				hurt = FALSE*/
 	if(hit_atom.density && isturf(hit_atom))
 		if(hurt)
-			Weaken(2 SECONDS)
-			take_organ_damage(10)
+			visible_message("<span class='danger'>[src] slams into [hit_atom]!</span>", "<span class='userdanger'>You slam into [hit_atom]!</span>")
+			take_organ_damage(7.5 * speed) // speed while thrower is standing still is 2, while walking with an aggressive grab is 2.4
+			KnockDown(2 SECONDS)
```
Comment (hal9000PR) [inline code/modules/mob/living/carbon/carbon.dm:596]: I would also considering increasing the knockdown values slightly I maybe something like 3 seconds?

File: code/modules/mob/living/carbon/carbon.dm (hunk 4 above)
Comment (Contrabang) [reply code/modules/mob/living/carbon/carbon.dm:604]: I was trying some stuff and going way over the top with stuff like `log((10 + speed))` but damage = 10 + 1.5 * speed is way better, will implement both of these

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**